src/otx/algo/classification/dino_v2.py

```diff
@@ -48,8 +50,16 @@ def __init__(
     ):
         super().__init__()
         self._init_args = get_class_initial_arguments()
+
+        repo_or_dir: str | Path = "facebookresearch/dinov2"
+        source: str = "github"
+        ci_data_root = os.environ.get("CI_DATA_ROOT")
+        if ci_data_root is not None:
+            repo_or_dir = Path(Path(ci_data_root) / "torch" / "hub" / "facebookresearch_dinov2_main")
+            source = "local"
         self.backbone = torch.hub.load(
-            repo_or_dir="facebookresearch/dinov2",
+            repo_or_dir=str(repo_or_dir),
```

Comment:
Did you do anything for the local DINOV2 repository here to modify the source URL of pretrained weights?
Even if you change the repo source but the repo source as no modification, it still tries to download the pretrained weights from the facebook storage server.
https://github.com/facebookresearch/dinov2/blob/e1277af2ba9496fbadf7aec6eba56e8d882d1e35/dinov2/hub/backbones.py#L57

Reply:
`source` would be changed from 'github' to 'local' when environment variable `CI_DATA_ROOT` which already used for specifying the storage mounted to the ci-runner container. I've copied pretrained weights there and torch.hub.load() will try to get them from there within our ci workflow. i've double checked that tests were failed when the storage does not have weights file.

Reply:
As @vinnamkim commented, current change uses remote URL to get the weights if the system cache doesn't have them. I will push new commits to load pretrained weights directly from the local storage instead of using torch.hub.load() API.